src/cryptoadvance/specter/api/rest/api.py

```diff
@@ -0,0 +1,149 @@
+import logging
+import json
+
+import cryptoadvance.specter as specter
+from cryptoadvance.specter.api.rest.base import (
+    AdminResource,
+    BaseResource,
+    SecureResource,
+    rest_resource,
+)
+from cryptoadvance.specter.api.security import require_admin, verify_password
+from flask_restful import reqparse
+from flask import current_app as app
+from datetime import datetime
+from ...specter_error import SpecterError
+
+from .. import auth
+
+logger = logging.getLogger(__name__)
+
+
+@rest_resource
+class ResourceSpecter(SecureResource):
+    """ /api/v1alpha/specter """
+
+    endpoints = ["/v1alpha/specter"]
+
+    def get(self):
+        specter_data = app.specter
+
+        return_dict = {
+            "data_folder": specter_data.data_folder,
+            "file_config": specter_data.file_config,
+            "config": specter_data.config,
+            "is_configured": specter_data._is_configured,
+            "is_running": specter_data._is_running,
+            "info": specter_data._info,
+            "network_info": specter_data._network_info,
+            "device_manager_datafolder": specter_data.device_manager.data_folder,
+            "devices_names": specter_data.device_manager.devices_names,
+            "wallets_names": specter_data.wallet_manager.wallets_names,
+            "last_update": datetime.now().strftime("%m/%d/%Y, %H:%M:%S"),
+        }
+
+        # Include alias list for easy lookup between names and alias
+        # Maybe there's a better way to create this
+        wallets_alias = []
+        alias_name = {}
+        name_alias = {}
+        for wallet in return_dict["wallets_names"]:
+            alias = specter_data.wallet_manager.wallets[wallet].alias
+            wallets_alias.append(alias)
+            alias_name[alias] = wallet
+            name_alias[wallet] = alias
+        return_dict["alias_name"] = alias_name
+        return_dict["name_alias"] = name_alias
+        return_dict["wallets_alias"] = wallets_alias
+        return json.dumps(return_dict)
+
+
+@rest_resource
+class ResourceTXlist(SecureResource):
+    """ /api/v1alpha/full_txlist """
+
+    endpoints = ["/v1alpha/full_txlist/"]
+
+    def get(self):
+        try:
+            validate_merkle_proofs = app.specter.config.get("validate_merkle_proofs")
+            idx = 0
+            tx_len = 1
+            tx_list = []
+            while tx_len > 0:
+                transactions = app.specter.wallet_manager.full_txlist(
```

Comment:
Note that the txlist code changed a lot, we also practically have already a semi API for it. I'd suggest to just move the part I marked as API over to this actual API, or if that's a problem I could do that, just let me know if you want.

Here's the part I referred to: https://github.com/cryptoadvance/specter-desktop/blob/75d3a1856c8782f57180c1c70215774f6e2bcfd0/src/cryptoadvance/specter/server_endpoints/wallets.py#L861

Reply:
@ben-kaufman I believe the wallet end point above and the txlist address most of the needs I envisioned for the API. Been testing it and works well. I'm going to suggest we close this PR.